libpod/runtime_pod.go

```diff
@@ -29,6 +29,13 @@ func (r *Runtime) NewPod(options ...PodCreateOption) (*Pod, error) {
 		return nil, errors.Wrapf(err, "error creating pod")
 	}
 
+	name, err := r.GenerateName()
```

Comment:
If you move this below the `for` loop over the options and add a check that `name == ""`like the container code is, we can be slightly more efficient in cases where the user passes in a name

Reply:
Fixed